### PR TITLE
Ports the current TG mechs to us, thus adding the Ripley MK I

### PR DESCRIPTION
**Pull request by Coolgat3**
[Changelogs]:

:cl: Coolgat3
add: Pretty much copied tg's robotics regarding the Ripley mech MK I. Now to get the actual normal ripley you use a conversion kit.
/:cl:

[why]: 
The ripley MK I is a really cool mech, it could be really useful for cargo or just meele fights, looking at the fact that it is a roundstart constructable mech that is pretty fast, it makes the ripley actually more useful and worth making.

Be advised I am not entirely sure if this won't change something we already have changed in robotics. I literally just copied the files from the PR made by ExcessiveUseOfCobblestone from TG which adds the ripley MKII.  Assuming this only mostly changes mech related things I'm really hoping it doesn't anyhow change what we already had changed or conflict with any other code. Honestly I'd check that, if I KNEW if there's anything breaking the stuff actually.

tgstation/tgstation#42828

### CODE REVIEW

**Review by deathride58 — DISMISSED** [review]
link original PR you're basing your changes off of in your PR with the `orgname/reponame#prnumhere` format

File: code/game/mecha/equipment/tools/work_tools.dm [hunk 1]
```diff
@@ -43,7 +43,7 @@
 					O.forceMove(chassis)
 					O.anchored = FALSE
 					occupant_message("<span class='notice'>[target] successfully loaded.</span>")
-					log_message("Loaded [O]. Cargo compartment capacity: [cargo_holder.cargo_capacity - cargo_holder.cargo.len]")
+					log_message("Loaded [O]. Cargo compartment capacity: [cargo_holder.cargo_capacity - cargo_holder.cargo.len]", LOG_MECHA)
```
Comment (deathride58) [inline code/game/mecha/equipment/tools/work_tools.dm]: you can revert all of these LOG_MECHA changes, they're the result of a small rework of how logs work on tg
keeping them around without also porting the logging rework results in compiling errors

File: code/game/mecha/mecha.dm [hunk 2]
```diff
@@ -127,15 +133,16 @@
...
+		add_airtank()
 	spark_system.set_up(2, 0, src)
 	spark_system.attach(src)
 	smoke_system.set_up(3, src)
 	smoke_system.attach(src)
 	add_cell()
 	START_PROCESSING(SSobj, src)
 	GLOB.poi_list |= src
-	log_message("[src.name] created.")
+	log_message("[src.name] created.", LOG_MECHA)
```
Comment (deathride58) [inline code/game/mecha/mecha.dm]: theres still more instances of LOG_MECHA

Comment (Coolgat3) [reply code/game/mecha/mecha.dm]: oh shit i thought it's only in one file i'll get onto that

**Comment by Coolgat3** [comment]
Alright LOG_MECHA stuff completely gone now.

**Comment by Poojawa** [comment]
ports using webeditor makes me sad

File: code/modules/research/techweb/all_nodes.dm [hunk 3]
```diff
@@ -8,9 +8,9 @@
 	display_name = "Basic Research Technology"
 	description = "NT default research technologies."
 	// Default research tech, prevents bricking
-	design_ids = list("basic_matter_bin", "basic_cell", "basic_scanning", "basic_capacitor", "basic_micro_laser", "micro_mani",
-	"destructive_analyzer", "circuit_imprinter", "experimentor", "rdconsole", "design_disk", "tech_disk", "rdserver", "rdservercontrol", "mechfab",
-	"space_heater", "xlarge_beaker", "sec_rshot", "sec_bshot", "sec_slug", "sec_Islug", "sec_dart", "sec_38", "sec_38lethal",
```
Comment (deathride58) [inline code/modules/research/techweb/all_nodes.dm:13]: Removal of sec_38lethal is an undocumented change

File: code/modules/research/techweb/all_nodes.dm [hunk 4]
```diff
@@ -8,9 +8,9 @@
...
+	design_ids = list("basic_matter_bin", "basic_cell", "basic_scanning", "basic_capacitor", "basic_micro_laser", "micro_mani", "desttagger", "handlabel", "packagewrap",
+	"destructive_analyzer", "circuit_imprinter", "experimentor", "rdconsole", "design_disk", "tech_disk", "rdserver", "rdservercontrol", "mechfab", "paystand",
+	"space_heater", "beaker", "large_beaker", "bucket", "xlarge_beaker", "sec_rshot", "sec_beanbag_slug", "sec_bshot", "sec_slug", "sec_Islug", "sec_dart", "sec_38",
```
Comment (deathride58) [inline code/modules/research/techweb/all_nodes.dm:13]: Additions of desttagger, handlabel, packagewrap, paystand (something that doesnt even exist), beaker, large_beaker, bucket, and sec_beanbag_slug are all undocumented changes

**Review by deathride58 — CHANGES_REQUESTED** [review]
Way, way too many undocumented changes in all_nodes.dm. You've reverted every single one of the node price changes, removed the requirement changes for bluespace research, and more. You've also added numerous references to designs that simply do not exist at all.

**Comment by silicons** [comment]
inactivity

**Pull request closed without merging**